### PR TITLE
Dev overview snippetize

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Deletes original overview section, moves intro paras from hypertables, caggs, and compression into snippets, and includes those snippets. Merge this in conjunction with the matching web- repo PR: https://github.com/timescale/web-documentation/pull/556

# Links

Fixes https://github.com/timescale/docs-private/issues/69

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

**Review by Rajakavitha1 — APPROVED** [review]
The edits LGTM!!!!

File: _partials/_compression-intro.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+When you enable compression, the data in your hypertable is compressed chunk by
```
Comment (charislam) [inline _partials/_compression-intro.md:1]: I'm really unsure about how this works as an intro. It reads more to me as "how compression works under the hood" than "what compression is and why you should use it".

Especially on the homepage, would a softer introduction to the topic work better? Is the first thing a reader wants to know about Timescale the shape of their compressed data? (It's very important later, just thinking about this in terms of the homepage)

File: _partials/_hypertables-intro.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Hypertables are PostgreSQL tables with special features that make it easy to
```
Comment (charislam) [inline _partials/_hypertables-intro.md:1]: On the other hand, this works great as an intro 😄 and on the homepage

File: _partials/_compression-intro.md (hunk 1 above)
Comment (Loquacity) [reply _partials/_compression-intro.md:1]: Yeah, I agree. I think we can improve this a lot.

**Pull request merged**